File: app/src/main/java/com/hold1/pagertabsdemo/fragments/TabsFragment.java
package com.hold1.pagertabsdemo.fragments;

import android.graphics.Color;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.RadioGroup;
import android.widget.SeekBar;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.hold1.pagertabsdemo.MainActivity;
import com.hold1.pagertabsdemo.R;
import com.hold1.pagertabsindicator.PagerTabsIndicator;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by Cristian Holdunu on 09/11/2017.
 */

public class TabsFragment extends Fragment implements FragmentPresenter {

    PagerTabsIndicator tabsIndicator;


    @BindView(R.id.text_size)
    SeekBar textSize;

    @BindView((R.id.tab_padding))
    SeekBar tabPadding;

    @BindView(R.id.tab_elevation)
    SeekBar tabElevation;

    @BindView(R.id.indicator_height)
    SeekBar indicatorHeight;

    @BindView(R.id.indicator_bg_height)
    SeekBar indicatorBgHeight;

    @BindView(R.id.indicator_margin)
    SeekBar indicatorMargin;

    @BindView(R.id.lock_expanded)
    CheckBox lockExpanded;

    @BindView(R.id.indicator_type)
    RadioGroup indicatorType;

    @BindView(R.id.add_dummy)
    Button addDummy;

    @BindView(R.id.show_bar_indicator)
    CheckBox showBarIndicator;

    @BindView(R.id.highlight_text)
    CheckBox highlightText;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.tabs_fragment, null);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        tabsIndicator = ((MainActivity) getActivity()).getTabsIndicator();
        if (tabsIndicator == null) return;
        ButterKnife.bind(this, view);

        textSize.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                tabsIndicator.setTextSize(progress);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {

            }
        });
        tabPadding.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                tabsIndicator.setTabPadding(progress);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {

            }
        });
        tabElevation.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                tabsIndicator.setTabElevation(progress);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {

            }
        });
        indicatorHeight.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                tabsIndicator.setIndicatorHeight(progress);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {

            }
        });

        indicatorBgHeight.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                tabsIndicator.setIndicatorBgHeight(progress);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {

            }
        });

        indicatorMargin.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                tabsIndicator.setIndicatorMargin(progress);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {

            }
        });

        lockExpanded.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                tabsIndicator.setLockExpanded(isChecked);
            }
        });

        indicatorType.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                if (checkedId == R.id.bar_top) {
                    tabsIndicator.setIndicatorType(PagerTabsIndicator.TAB_INDICATOR_TOP);
                } else {
                    tabsIndicator.setIndicatorType(PagerTabsIndicator.TAB_INDICATOR_BOTTOM);
                }
            }
        });
        addDummy.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ((MainActivity) getActivity()).addDummyTab();
            }
        });

        showBarIndicator.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                tabsIndicator.setShowBarIndicator(isChecked);
            }
        });

        highlightText.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                tabsIndicator.setHighlightText(true);
                tabsIndicator.setHighlightTextColor(Color.RED);
            }
        });
    }

    @Override
    public String getTabName() {
        return "Tabs";
    }

    @Override
    public int getTabImage() {
        return R.drawable.ic_tabs;
    }

    @Override
    public String getTabImageUrl() {
        return "https://s3-us-west-2.amazonaws.com/anaface-pictures/ic_tab.png";
    }
}

File: app/src/main/java/com/hold1/pagertabsdemo/fragments/TabsFragment.kt
package com.hold1.pagertabsdemo.fragments

import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.SeekBar
import android.widget.SeekBar.OnSeekBarChangeListener
import androidx.fragment.app.Fragment
import com.hold1.pagertabsdemo.ViewPagerActivity
import com.hold1.pagertabsdemo.R
import com.hold1.pagertabsdemo.databinding.TabsFragmentBinding
import com.hold1.pagertabsindicator.PagerTabsIndicator

/**
 * Created by Cristian Holdunu on 09/11/2017.
 */
class TabsFragment : Fragment(), FragmentPresenter {
    var tabsIndicator: PagerTabsIndicator? = null

    private lateinit var binding: TabsFragmentBinding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        binding = TabsFragmentBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        tabsIndicator = (activity as? ViewPagerActivity)?.tabsIndicator
        binding.textSize.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                tabsIndicator?.textSize = progress
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })
        binding.tabPadding.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                tabsIndicator?.tabPadding = progress
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })
        binding.tabElevation.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                tabsIndicator?.tabElevation = progress
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })
        binding.indicatorHeight.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                tabsIndicator?.indicatorHeight = progress
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })
        binding.indicatorBgHeight.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                tabsIndicator?.indicatorBgHeight = progress
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })
        binding.indicatorMargin.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                tabsIndicator?.indicatorMargin = progress
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}
            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })
        binding.lockExpanded.setOnCheckedChangeListener { _, isChecked -> tabsIndicator?.isLockExpanded = isChecked }
        binding.indicatorType.setOnCheckedChangeListener { _, checkedId ->
            if (checkedId == R.id.bar_top) {
                tabsIndicator?.indicatorType = PagerTabsIndicator.TAB_INDICATOR_TOP
            } else {
                tabsIndicator?.indicatorType = PagerTabsIndicator.TAB_INDICATOR_BOTTOM
            }
        }
        binding.addDummy.setOnClickListener { (activity as? ViewPagerActivity)?.addDummyTab() }
        binding.showBarIndicator.setOnCheckedChangeListener { _, isChecked -> tabsIndicator?.showBarIndicator = isChecked }
        binding.highlightText.setOnCheckedChangeListener { _, isChecked ->
            tabsIndicator?.isHighlightText = isChecked
            tabsIndicator?.highlightTextColor = Color.RED
        }
    }

    override val tabName: String
        get() = "Tabs"
    override val tabImage: Int
        get() = R.drawable.ic_tabs
    override val tabImageUrl: String
        get() = "https://s3-us-west-2.amazonaws.com/anaface-pictures/ic_tab.png"
}